ASCIT/donut
# Room reservations

### Summary
- Added room reservations

### Test Plan
- Try booking some rooms; make sure conflicts are checked correctly
- Make sure they show up under "my reservations"
- Execute some test queries under "all reservations"
- Make sure "room info" is displaying correctly
